go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -803,11 +799,16 @@ func isLayer2UserDefinedPrimaryNetwork(netInfo util.NetInfo) bool {
 	return netInfo.IsPrimaryNetwork() && netInfo.TopologyType() == types.Layer2Topology
 }
 
-func autoExcludeCIDRs(subnet *net.IPNet) []*net.IPNet {
-	gwIP := util.GetNodeGatewayIfAddr(subnet).IP
-	mgmtPortIP := util.GetNodeManagementIfAddr(subnet).IP
-	return []*net.IPNet{
-		{IP: gwIP, Mask: util.GetIPFullMask(gwIP)},
-		{IP: mgmtPortIP, Mask: util.GetIPFullMask(mgmtPortIP)},
+func infrastructureExcludeCIDRs(netInfo util.NetInfo) []*net.IPNet {
+	var excludeCIDRs []*net.IPNet
+
+	for _, subnet := range netInfo.Subnets() {
+		gwIP := netInfo.GetNodeGatewayIP(subnet.CIDR).IP
```

Comment:
ah ok so if the user has set custom default gateway IP, then that's what we exclude from IPAM, gotcha.